metriq_gym/job_manager.py

```diff
@@ -0,0 +1,59 @@
+import json
+import uuid
+from typing import List, Dict, Any
+
+DEFAULT_FILE_PATH = ".metric_gym_jobs.jsonl"
+
+# TODO: JobManager is not thread-safe at the moment
+
+
+class JobManager:
```

Comment:
We may also want to create a method here called `get_job_result` that consolidates the Qiskit and Quantinuum job result extraction that currently live in `process.py`.